### DISCUSSION TITLE
Current build and installation instructions for lilac?

### DISCUSSION THREAD

**Discussion by Pazu** [petefoth/unofficial-l4m-builds#22] [General]
Hi, Pete. Are you by any chance still working on this project?

Do you have any builds newer than the lineage-20.0-20231008-4microg-UNOFFICIAL-lilac.zip available from [here](https://androidfilehost.com/?fid=10620683726822085936)?

Then, do you have any basic installation instructions for it? I'd really like to set up my old lilac as a back-up device for when my XZ2 Compact's display gives up the ghost for the second time, which I suspect will be soon based on what I'm seeing (not going to replace it).

**Comment by petefoth** [comment]
Hi Pazu
I am still working on it, and still making builds - unofficial and 'semi-official' (see **Note 1**) for a couple of custom ROMs
- [LineageOS for Microg](https://lineage.microg.org/)
   - 'semi-official' 20.0 build available at https://download.lineage.microg.org/lilac/ (see **Note 2**)
   - unofficial 21.0 builds available at https://sourceforge.net/projects/pf-custom-roms/files/lilac/los4microg/
- [IodéOS](https://iode.tech/iodeos-en/) 
   - unofficial Iodé 4.10 https://sourceforge.net/projects/pf-custom-roms/files/lilac/IodeOS/ (see **Note 2**)

See [this XDA Forum thread](https://xdaforums.com/t/rom-lineageos-21-0-unofficial-1-0-2024-02-25.4658694/) and [this thread in the IodéOS Community forums](https://community.iode.tech/t/unofficial-build-for-sony-xperia-zx1-compact-lilac/1097) for any questions, and news of new builds. 

**Note 1.** They are 'semi-official' because lilac is not officially supported by LineageOS but, as a project maintainer and contributor to the LineageOS fro MicroG, I am able to make los4microG builds using the project build server, signed with the project keys, and updateable over-the-air.

**Note 2.** Android released a [set of changes in April](https://source.android.com/docs/whatsnew/android-14-release) which 'broke` many custom ROMs. ROM maintainers had to make significant changes to their code before their ROMs would build with the changed Android code. The breaking changes appeared after the  unofficial los4microG 21.0  build, but before I had time to make semi-official los4microG and IodéOS 5 builds. If / when the maintainer of  the unofficial LineageOS makes the necessary changes to allow the code to build with the Android changes I will attempt to make  new los4microG 21.0 and IodéOS 5 builds.

**Comment by petefoth** [comment]
> do you have any basic installation instructions for it?

See this post

https://community.iode.tech/t/installing-unofficial-build-for-xperia-xz1-compact-lilac/1494

**Reply by Pazu** [reply]
Thanks very much! Do those instructions also apply to the semi-official LineageOS for Microg build? Those are probably what I'd be interested in.

**Reply by petefoth** [reply]
Yes they should do. Just flash the LineageOS for Microg .zip file instead of the IodéOS one.

The only thing I would caution about using the semi-official build is that, I don't know how well upgrading to a build of a new android version (e.g. 20.0 -> 21.0) will work if / when it happens:
- OTA upgrade will probably not work - you will probably have to flash manually using TWRP 
- no guarantee that user-installed apps and data will survive a manual upgrade

I currently use my unofficial 21.0 build, and I plan to manually flash the semi-official build when it arrives. I ***think*** that will go OK, but I'll backup my user-installed apps and data before I update, just in case

**Comment by Pazu** [comment]
Might I ask, how are these three builds as far as "SafetyNet" or "Root Beer" or whatever the current iteration of that stuff goes?

**Reply by petefoth** [reply]
The los4microG builds don't pass SafetyNet. The IodéOS build does, at least it did last time I tried it.

I don't know about RootBeer. None of the builds are rooted (if that's what RootBeer checks for) so they should be OK

**Reply by Pazu** [reply]
> The los4microG builds don't pass SafetyNet. The IodéOS build does, at least it did last time I tried it.

Hm, that changes things. I might well need SafetyNet for banking apps.

I've been an e/OS/ user for some time on my XZ2c (currently running 2.0) and that, I believe, passes SN, and all the apps I want to use certainly work well on it. (Too bad there's no e/OS/ for XZ1c any more, but I understand you didn't really have an interest in keeping it up.) Although, if I recall correctly, those e/OS/ builds you may for XZ1c did _not_ pass SN, did they?

Thus it seems likely the Iodé build may suit me best.

**Reply by petefoth** [reply]
> Hm, that changes things. I might well need SafetyNet for banking apps.

I use banking apps on los4microG. The ones I use are all from the UK: Smile, Nationwide, & Triodos. Many banking apps work fine without SafetyNet. So I wouldn't write off los4microg immediately.

> Too bad there's no e/OS/ for XZ1c any more, but I understand you didn't really have an interest in keeping it up

Mostly lack of time, and other demands on my time. /e/OS was my first experience of privacy-respecting custom ROMs (and of building them), but they made some design and implementation decisions that I strongly disagreed with, and refused to reconsider them. One good thing about being a volunteer rather than a paid employee is that you can walk away from bosses who do the wrong thing 😄

**Comment by Pazu** [comment]
Pete, I finally got around to installing your unofficial (semi-official now?) LOS 21 build of 2024-06-20 from download.lineage.microg.org/lilac/. Thanks for making it, it's working well on my XZ1c. I guess that means the necessary changes you mentioned got done.

**Reply by petefoth** [reply]
Yes the changes got made in the code used by lilac so that it builds with the  Android changes made in April. 

Android make platform releases every three months, referred to as 'Quarterly Platfrom Releases), so the APril changes were 'QPR2'. They have recently released QPR3 which also has some breaking changes which need  work to allow LineaegOS (and los4microG and IodéOS) to build. 

The maintainer of the lilac code knows about this, and it is on their 'To Do' list. So there won't be updated builds until that gets done. It ***may*** get done in time for the next batch of semi-official builds which are due towards the end of the month. Whenever it gets done, the update should show up in the Updater app on your phone, which should show a notification when an update is available.

**Comment by Pazu** [comment]
I did find one banking app wouldn't run on the LineageOS build. Do you think either of the Iodé builds would fare better in that respect?

**Reply by petefoth** [reply]
It's possible: IodéOS builds usually pass SafetyNet, and the los4microG builds don'e. If your bank app is relying on SafetyNet, then it ***may*** work with IodéOS. The only way to find out is to try 😄

**Comment by Pazu** [comment]
May I ask, what's the difference between "iode-5.1-20240617-lilac.zip" and "iode-5.1-20240617-UNOFFICIAL-lilac-img.zip"?

**Reply by petefoth** [reply]
`...lilac.zip`is the main ROM zip file which is flashed using recovery (e.g `adb sideload`)
`...lilac-img.zip` is used when installing  / updating with `fastboot` in botloadeer mode (e.g. `fastboot --skip-reboot  update iode-5.1-20240617-UNOFFICIAL-lilac-img.zip` 

Using recovery is the 'normal' method. Using `fastboot` (wrapped in a `flash-all.sh` or `flash-all.bat` script) is the official way of installing official builds - see [this forum thread](https://community.iode.tech/t/new-installation-method/1383) for more information.

**Reply by Pazu** [reply]
Ah, okay, thanks. I had previously flashed your lineage-21.0-20240620-microG-UNOFFICIAL-lilac-recovery.img. I presume that's fine to continue using for sideloading Iodé?

**Reply by petefoth** [reply]
It should be

**Comment by Pazu** [comment]
The sideload operation (% adb sideload iode-5.1-20240617-lilac.zip) reaches 47% and the phone display shows:
```
ERROR: recovery: Signature verification failed
ERROR: recovery: error:21
```
Is that okay?

Followup: seems to be okay after installing anyway. I had also tried installing from the SD card and got the same error, though.

**Reply by petefoth** [reply]
The 47% thing is normal: it means that the ROM zip file has successfully been uploaded from the computer to the phone, so the computer has done it's work and the rest of the job will be done on the phone. If the phone eventually reports that the install has succeeded, then the computer ***may*** display 'Complete'. But it's also quite normal for that not to happen. Generally speaking 47% is fine :)

The signature verification error is expected because you are installing my unofficial Iodé build, signed with my keys, over the 'semi-official' los4microG build which is signed with the los4microG project keys. As you have discovered, it is safe to ignore the error and carry on installing the build.

**Comment by Pazu** [comment]
Thanks very much. Indeed, the problematic banking app works fine on IodéOS.

**Reply by petefoth** [reply]
That's great. Which banking app is it?

**Reply by Pazu** [reply]
[George Österreich](https://play.google.com/store/apps/details?id=at.erstebank.george).

**Comment by Pazu** [comment]
FWIW, the screenshot functionality appears not to work. Screenshots are just solid black.

Also I noticed that Authy doesn't work. It shows an error, "Invalid request: Integrity token param is required." I'll ask about that in the OS's own Community.
